crates/control/src/role_assignment.rs

```diff
@@ -97,9 +98,9 @@ impl RoleAssignment {
         {
             role = match context.player_number {
                 PlayerNumber::One => Role::Keeper,
-                PlayerNumber::Two => Role::DefenderRight,
-                PlayerNumber::Three => Role::StrikerSupporter,
-                PlayerNumber::Four => Role::DefenderLeft,
+                PlayerNumber::Two => context.optional_roles[0],
+                PlayerNumber::Three => context.optional_roles[1],
+                PlayerNumber::Four => context.optional_roles[2],
```

Comment:
This might panic, if the `Vec` is too short. Consider using `Vec::get(...).unwrap_or(...)` https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get

Reply:
done